pkg/api/types.go

```diff
@@ -308,6 +308,18 @@ type KubernetesConfigDeprecated struct {
 	CtrlMgrRouteReconciliationPeriod string `json:"ctrlMgrRouteReconciliationPeriod,omitempty"`
 }
 
+// KubeProxyMode is for iptables and ipvs (and future others)
+type KubeProxyMode string
+
+// We currently support ipvs and iptables
+const (
+	KubeProxyModeIPTables KubeProxyMode = "iptables"
+	KubeProxyModeIPVS     KubeProxyMode = "ipvs"
+)
+
+// DefaultKubeProxyMode is the default KubeProxyMode value
+const DefaultKubeProxyMode KubeProxyMode = KubeProxyModeIPTables
```

Comment:
I don't think we should define defaults in here. Defaults should go in https://github.com/Azure/aks-engine/blob/master/pkg/api/const.go